alerting/src/main/kotlin/org/opensearch/alerting/alerts/AlertMover.kt

```diff
@@ -116,3 +127,108 @@ private fun alertContentParser(bytesReference: BytesReference): XContentParser {
     XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, xcp.nextToken(), xcp)
     return xcp
 }
+
+/**
+ * Moves defunct active alerts to the alert history index when the corresponding workflow or trigger is deleted.
+ *
+ * The logic for moving alerts consists of:
+ * 1. Find active alerts:
+ *      a. matching workflowId if no workflow is provided (postDelete)
+ *      b. matching workflowid and chained alert triggerIds if monitor is provided (postIndex)
+ * 2. Move alerts over to DataSources.alertsHistoryIndex as DELETED
+ * 3. Delete alerts from monitor's DataSources.alertsIndex
+ * 4. Schedule a retry if there were any failures
+ */
+suspend fun moveAlerts(client: Client, workflowId: String, workflow: Workflow?, monitorCtx: MonitorRunnerExecutionContext) {
```

Comment:
Why dont we just modify the original function to support both use cases and we can add helper functions to keep it simple.

Reply:
the checks are different.

- we first have to resolve datasources for correct index name.

- the check in postIndex(), postDeleete() for isAlertIndicesInitialized is applied before running move alerts for monitors and after we resolve it for workflows. the short circuiting happens at different places. have to use different field names in search request for alert docs.

will add a follow up PR to extract common logic snippets like moving alerts to history  index and deleting from alert index